### PR TITLE
Added deleteDir to async list also.

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

**Comment by mickel1138** [comment]
The ZIP file will not delete, because it's owned by Pelagos, we should probably modify the deleteDir worker to also delete the zip file. And rename the worker to something like DeleteDatasetFiles or something.

**Review by fightingtexasaggie — APPROVED** [review]
confirmed after this change the deleteDir call is now running true async, under user pelagos, not apache.

File: src/Event/DatasetSubmissionListener.php [hunk 1]
```diff
@@ -214,8 +214,24 @@ public function onAcceptReview(EntityEvent $event)
...
-        $this->messageBus->dispatch($deleteFileMessage);
+        $fileset = $datasetSubmission->getFileset();
+
+        if (!$fileset instanceof Fileset) {
+            $this->logger->warning('No files exist in this dataset.');
+            return;
+        }
+
+        foreach ($fileset->getDeletedFiles() as $deletedFile) {
+            try {
```
Comment (mickel1138) [inline src/Event/DatasetSubmissionListener.php:225]: Why try? This is async!

File: src/MessageHandler/DeleteFileHandler.php [hunk 2]
```diff
@@ -56,33 +54,10 @@ public function __construct(LoggerInterface $deleteFileLogger, Datastore $datast
...
-
-        $this->logger->info(sprintf('Processing Dataset Submission with ID: "%s"', $datasetSubmissionId));
-
-        $datasetSubmission = $this->entityManager->getRepository(DatasetSubmission::class)->find($datasetSubmissionId);
-
-        if (!$datasetSubmission instanceof DatasetSubmission) {
-            $this->logger->warning('Dataset Submission was not found.');
-            return;
+        try {
+            $this->datastore->deleteFile($deletedFile->getFilePath());
```
Comment (mickel1138) [inline src/MessageHandler/DeleteFileHandler.php:58]: Check for succes.

Comment (praneethpr) [reply src/MessageHandler/DeleteFileHandler.php:58]: Added it in the datastore

**Review by praneethpr — APPROVED** [review]

**Pull request merged**